File: linear search.js
function linearsearch(arr, target)
{
  for (let i = 0; i < arr.length; i++)
  {
    if (arr[i] === target)
    {
      return i; //return th index if found
    }
  }
  return -1;
}

const arr = [10, 20, 30, 40, 50, 60];
const target = 40;

const result = linearsearch(arr, target);
if (result !== -1)
{
  console.log(`Element found at index ${result}`);
}
else
{
  console.log("Element not found");
}

File: Recursion/LinearSearch.java
package Recursion;

public class LinearSearch {

  public static void main(String[] args) {
    int[] arr = {-15, -5, 0, 1, 2, 6, 8, 11, 15};
    int target = 11;
    System.out.println(search(arr, target, 0));
    System.out.println(search1(arr, target, 0));
  }
  

  // return the index of element if it is present in the array
  static int search(int[] arr, int target, int index){

    if(index == arr.length){
      return -1;
    }

    if(arr[index] == target){
      return index;
    } else {
      return search(arr, target, index + 1);
    }
    
  }

  // return true/false based on presence of target element in the array
  static boolean search1(int[] arr, int target, int index){

    if(index == arr.length){
      return false;
    }

   return arr[index] == target || search1(arr, target, index + 1);
  }
}
